src/main/java/org/radarbase/appserver/service/scheduler/MessageSchedulerService.java

```diff
@@ -169,7 +169,7 @@ public void deleteScheduled(Message message) {
         schedulerService.deleteScheduledJob(key);
     }
 
-    public void sendMessage(FcmDownstreamMessage message) throws Exception {
+    public void send(FcmDownstreamMessage message) throws Exception {
```

Comment:
You can make the `send` method here `abstract`. So you can override it in the sub classes. This will enable you to use the full potential of inheritance.